src/Microsoft.VisualStudio.ProjectSystem.Managed/Text/StringBuilderExtensions.cs

```diff
@@ -46,5 +46,24 @@ public static void AppendFormat(this StringBuilder builder, StringFormat format)
                 builder.AppendFormat(format.Format, format.Arguments);
             }
         }
+
+        public static StringBuilder TrimEnd(this StringBuilder builder, params char[] trimChars)
+        {
+            while (builder.Length > 0)
+            {
+                foreach (var c in trimChars)
+                {
+                    if (builder[builder.Length - 1] == c)
+                    {
+                        builder.Length--;
+                        break;
+                    }
+
+                    return builder;
```

Comment:
This return seems wrong.  It unconditionally returns from function the first time a value in `trimChars` is not the end of the build.r

Reply:
Yes, those are the semantics of TrimEnd - remove characters from the end of the string that are one of `trimChars`.

Reply:
Consider `"hello".TrimEnd("zo")`. That produces `"hell". Your change though will produce `"hello"` because you unconditionally return after `z` doesn't match `o`.

Reply:
Yep.  Thanks, I thought I had that, but clearly not.

Reply:
Added an explicit check for a match.